frontend/src/lib/modals/sns/neurons/FollowSnsNeuronsByTopicStepTopics.svelte

```diff
@@ -0,0 +1,141 @@
+<script lang="ts">
+  import Separator from "$lib/components/ui/Separator.svelte";
+  import { i18n } from "$lib/stores/i18n";
+  import type { TopicInfoWithUnknown } from "$lib/types/sns-aggregator";
+  import FollowSnsNeuronsByTopicItem from "$lib/modals/sns/neurons/FollowSnsNeuronsByTopicItem.svelte";
+  import { fromDefinedNullable } from "@dfinity/utils";
+  import TooltipIcon from "$lib/components/ui/TooltipIcon.svelte";
+  import type { SnsTopicFollowing, SnsTopicKey } from "$lib/types/sns";
+  import {
+    getSnsTopicInfoKey,
+    snsTopicToTopicKey,
+  } from "$lib/utils/sns-topics.utils";
+  import TestIdWrapper from "$lib/components/common/TestIdWrapper.svelte";
+  import type { SnsNeuronId } from "@dfinity/sns";
+
+  type Props = {
+    topicInfos: TopicInfoWithUnknown[];
+    selectedTopics: SnsTopicKey[];
+    followings: SnsTopicFollowing[];
+    closeModal: () => void;
+    openNextStep: () => void;
+    removeFollowing: (args: {
+      topicKey: SnsTopicKey;
+      neuronId: SnsNeuronId;
+    }) => void;
+  };
+  let {
+    topicInfos,
+    selectedTopics = $bindable(),
+    followings,
+    closeModal,
+    openNextStep,
+    removeFollowing,
+  }: Props = $props();
+
+  let criticalTopicInfos: TopicInfoWithUnknown[] = $derived(
+    topicInfos.filter((topicInfo) => fromDefinedNullable(topicInfo.is_critical))
+  );
+  let nonCriticalTopicInfos: TopicInfoWithUnknown[] = $derived(
+    topicInfos.filter(
```

Comment:
very personal opinion
```suggestion
  topicInfos
      .map((topicInfo) => fromDefinedNullable(topicInfo.is_critical))
      .filter(isCritical => !isCritical))
```